### PR TITLE
Feat/GitHub action master

### LINKED ISSUE #22: IntegrationTest - GithubActions
Currently we have github actions in place, however integration tests are NOT included.

In the past we tried to include integration tests in the job, however they were behaving inconsistently:

```
[DEBUG] integration version: 0.0.0 commit: 8a9a2e535227768fdf5df6c16b28158d3f945484
[DEBUG] exporter msg=Initializing SWbemServices source=exporter.go:248
[INFO] exporter msg=Enabled collectors: service, cs source=exporter.go:326
[INFO] exporter msg=Starting windows_exporter (version=0.12.0-10-geff5f24-dirty, branch=HEAD, revision=eff5f2415398d89173a2203c47366e4882f3bc0e) source=exporter.go:351
[INFO] exporter msg=Build context (go=go1.14.4, user=fv-az44\runneradmin@fv-az44, date=20200603-16:55:13) source=exporter.go:352
[INFO] exporter msg=Starting server on 127.0.0.1:9183 source=exporter.go:355
[DEBUG] exporter msg=Generated WMI query SELECT * FROM Win32_ComputerSystem source=wmi.go:24
[DEBUG] exporter msg=Generated WMI query SELECT * FROM Win32_Service WHERE Name like '%' source=wmi.go:47
[DEBUG] exporter msg=collector cs succeeded after 0.314972s. source=exporter.go:209
DEBUG
len stdout: 0
DEBUG

    TestIntegration: integration_test.go:86: 
        	Error Trace:	integration_test.go:86
        	Error:      	Received unexpected error:
        	            	error validating, error: EOF
        	Test:       	TestIntegration
        	Messages:   	The output of WinServices integration doesn't have expected format
```

Same job passed in a re-run

### PR DESCRIPTION
**Pull request by paologallinaharbur**
- Included integration tests in github actions
 - solved minor issues
 - added more logs in debug mode

Fix #22

### CODE REVIEW

File: src/exporter/exporter.go [hunk 1]
```diff
@@ -115,7 +116,7 @@ func (e *Exporter) createJobObject() error {
 		return fmt.Errorf("failed to set job object info:%v", err)
 	}
 
-	windows.AssignProcessToJobObject(e.jobObject, (*process)(unsafe.Pointer(e.cmd.Process)).handle)
+	err = windows.AssignProcessToJobObject(e.jobObject, (*process)(unsafe.Pointer(e.cmd.Process)).handle)
```
Comment (paologallinaharbur) [inline src/exporter/exporter.go:119]: we were checking twice the same error

File: src/scraper/fetcher.go [hunk 2]
```diff
@@ -18,27 +21,35 @@ func Get(client HTTPDoer, url string) (MetricFamiliesByName, error) {
 		return mfs, err
 	}
 	req.Header.Set("Content-Type", "application/json")
+	log.Debug("Performing HTTP request against: %s", url)
+	t := time.Now()
 	resp, err := client.Do(req)
 	if err != nil {
 		return mfs, err
 	}
+	if resp.StatusCode != 200 {
```
Comment (paologallinaharbur) [inline src/scraper/fetcher.go:30]: we were not chcking the rest status code

File: src/winservices.go [hunk 3]
```diff
@@ -81,28 +90,40 @@ func run(e *exporter.Exporter, i *integration.Integration, interval time.Duratio
...
 			if err = nri.ProcessInventory(i); err != nil {
 				return fmt.Errorf("fail to process inventory:%v", err)
 			}
+			log.Debug("Inventory processed, time elapsed: %s", time.Since(t).String())
+
 			err = i.Publish()
 			if err != nil {
 				log.Error("failed to publish integration:%v", err)
 			}
+			log.Debug("Metrics and inventory published")
```
Comment (gsanchezgavier) [inline src/winservices.go:123]: I would remove this one since the published json is seen in the output

File: src/winservices.go [hunk 4]
```diff
@@ -81,28 +90,40 @@ func run(e *exporter.Exporter, i *integration.Integration, interval time.Duratio
...
 
 		case <-e.Done:
+			log.Debug("The exporter is not running anymore, the integration is going to be stopped")
```
Comment (gsanchezgavier) [inline src/winservices.go:126]: also this one , is duplicated with the log fatal when return

File: src/winservices.go [hunk 5]
```diff
@@ -58,20 +58,29 @@ func main() {
 	interval, err := time.ParseDuration(args.ScrapeInterval)
 	if err != nil {
 		log.Error(err.Error())
+		interval = minScrapeInterval
```
Comment (gsanchezgavier) [inline src/winservices.go:61]: oh good catch

File: src/winservices.go (hunk 3 above)
Comment (gsanchezgavier) [reply src/winservices.go:123]: I will change it when i rebase my pr if you agree

**Pull request merged**